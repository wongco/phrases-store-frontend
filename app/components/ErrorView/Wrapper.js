import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  margin: 1em 0;
`;

export default Wrapper;
