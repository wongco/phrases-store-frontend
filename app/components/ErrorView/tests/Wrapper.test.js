import React from 'react';
import { shallow, mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import toJson from 'enzyme-to-json';

import Wrapper from '../Wrapper';

describe('<Wrapper />', () => {
  it('renders without crashing', () => {
    shallow(<Wrapper />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<Wrapper />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render children properly', () => {
    const wrapper = mount(
      <Wrapper>
        <div>Hello!</div>
      </Wrapper>,
    );
    const renderedComponent = enzymeFind(wrapper, Wrapper);
    const divs = renderedComponent.find('div');
    expect(divs).toHaveLength(2);
    expect(divs.last().text()).toEqual('Hello!');
  });
});
