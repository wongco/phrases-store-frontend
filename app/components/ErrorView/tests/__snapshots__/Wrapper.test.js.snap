// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> matches snapshot for default criteria 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Wrapper-snmkqo-0",
        "isStatic": true,
        "rules": Array [
          "display:flex;flex-direction:row;justify-content:center;align-items:center;margin:1em 0;",
        ],
      },
      "displayName": "Wrapper",
      "render": [Function],
      "styledComponentId": "Wrapper-snmkqo-0",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
